src/Estimators/tests/RecursiveLeastSquareTest.cpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file RecursiveLeastSquareTest.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <cmath>
+#include <functional>
+#include <random>
+
+// Catch2
+#include <catch2/catch.hpp>
+
+// YARP
+#include <yarp/os/Bottle.h>
+#include <yarp/os/Property.h>
+#include <yarp/os/ResourceFinder.h>
+#include <yarp/os/Searchable.h>
+#include <yarp/os/Value.h>
+
+#include <iDynTree/Core/EigenHelpers.h>
+#include <iDynTree/Core/MatrixDynSize.h>
+
+#include <BipedalLocomotionControllers/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotionControllers/ParametersHandler/YarpImplementation.h>
+
+#include <BipedalLocomotionControllers/Estimators/RecursiveLeastSquare.h>
+
+#include <ConfigFolderPath.h>
+
+using namespace BipedalLocomotionControllers::Estimators;
+using namespace BipedalLocomotionControllers::ParametersHandler;
+
+/**
+ * The model class implement a simple model used for testing the Recursive Least Square algorithm
+ * |y1|   |x      x * x |   |p1|   |noise_1 |
+ * |  | = |             | * |  | + |        |
+ * |y2|   |sin(x) cos(x)|   |p2|   |noise_2 |
+ */
+class Model
+{
+    iDynTree::Vector2 params;
+    double x;
+
+    std::random_device rd{};
```

Comment:
Please never  use actual random sources in testing, as they affect repeatability of the tests. How would you debug if the test fails one? If you really want to do different tests every-time, generate a random integer, **print it on the tests log (so that you can reproduce the test later**, and then use  it as a seed for a (deterministic) pseudo random number generator.

Reply:
I can generate the noise offline. Store the values in the configuration file and use that. What do you think? in this case the test will be deterministic

Reply:
You could use the ``TestUtilities`` in iDynTree. If I remember well, they use a constant feed so that it is random but repeatable. Right @traversaro?

Reply:
`TestUtilities` in `iDynTree` generates a random variable that cannot be described with a Gaussian Distribution, right?

Reply:
> If getRandomDouble returns a number sampled from a uniform distribution I can use the Central limit theorem to generate a variable extracted from Gaussian Distribution.

> Otherwise, I use a seed that should generate a determinist pseudo-random sequence.
(#23 (comment))

Both are good options!

Reply:
I'll go for a seed.

Reply:
Fixed seed added in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/23/commits/54c264f5369702d5275b57501530dc472b45131c

Reply:
Yes, I was actually thinking  about the seed that is by far an easier solution.  Note that if you pass a seed to the `mt19937 ` generator, you can remove the `std::random_device` altogether.

Reply:
Thanks @traversaro. I'll change it immediately

Reply:
Done in 0260479863d7c92a4082fb295b3da9031a6a368e